File: backend/Watchdog.Core/Watchdog.Core.Common/Models/Issue/IssueMessage.cs
using Newtonsoft.Json;
using System;
using Watchdog.Core.Common.DTO.Application;

namespace Watchdog.Core.Common.Models.Issue
{
    public class IssueMessage
    {
        public string Id { get; set; }
        public int IssueId { get; set; }
        public string ApiKey { get; set; }

        [JsonProperty("project")]
        public ApplicationDto Application { get; set; }

        public DateTime OccurredOn { get; set; }
        public string Platform => IssueDetails.EnvironmentMessage.Platform;
        public string Browser => IssueDetails.EnvironmentMessage.Browser;
        public IssueMessageDetails IssueDetails { get; set; }
    }
}

File: frontend/src/app/shared/models/issue/issue-message.ts
import { IssueDetails } from '@shared/models/issue/issue-details';
import { Project } from '../projects/project';

export interface IssueMessage {
    id?: string,
    issueId?: number,
    occurredOn: Date,
    issueDetails: IssueDetails,
    project: Project
}
